### PR TITLE
RBAC: Cover plugin routes

### PR DESCRIPTION
**Pull request by gamab**
**What is this feature?**

This PR adds the possibility to protect plugin proxied routes with an RBAC action check.

**Why do we need this feature?**

This has been requested by plugin developers.

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

**Special notes for your reviewer:**

We had covered in https://github.com/grafana/grafana/pull/57582 includes.

Schema updates: https://github.com/grafana/grafana/pull/80592

Once this is merged I need to update the plugin-tools docs

Please check that:
- [x] It works as expected from a user's perspective.
- [x] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/plugins/plugindef/plugindef.cue [hunk 1]
```diff
@@ -364,6 +364,9 @@ schemas: [{
 			reqSignedIn?: bool
 			reqRole?:     string
 
+			// RBAC action the user must have to access the route
+			action?: string
```
Comment (gamab) [inline pkg/plugins/plugindef/plugindef.cue:368]: Do we prefer `reqAction` to be consistent with `reqRole` or `action` to be consistent with the includes' `action` field?

Comment (andresmgot) [reply pkg/plugins/plugindef/plugindef.cue:368]: I'd go with `reqAction` to make it clear it's a requirement. Also an example will help for people to understand what this does.

Unrelated but does this support multiple actions or only one?

**Review by andresmgot — COMMENTED** [review]
Code LGTM, I only have a suggestion regarding naming / docs

File: pkg/plugins/plugindef/plugindef.cue (hunk 1 above)
Comment (gamab) [reply pkg/plugins/plugindef/plugindef.cue:368]: Ok :ok_hand:

> Also an example will help for people to understand what this does.
What do you have in mind? Plugin example?

It supports only one action. We'll see if we ever need to support something more complex.

Comment (andresmgot) [reply pkg/plugins/plugindef/plugindef.cue]: > What do you have in mind? Plugin example? Or plugin-tools docs update with an example?
For this case, I only mean in the comment: `// RBAC action the user must have to access the route i.e. plugin-id.projects:read`

> It supports only one action
:+1:

**Review by andresmgot — APPROVED** [review]
LGTM

**Pull request merged**